core/src/main/scala/org/apache/spark/util/JsonProtocol.scala

```diff
@@ -100,7 +102,16 @@ private[spark] object JsonProtocol {
         executorMetricsUpdateToJson(metricsUpdate)
       case blockUpdate: SparkListenerBlockUpdated =>
         blockUpdateToJson(blockUpdate)
-      case _ => parse(mapper.writeValueAsString(event))
+      case _ =>
+        val outputStream = new PipedOutputStream()
```

Comment:
Hi @brkyvz, how about adding a note that `PipedOutputStream` is intentionally used to get rid of additional consumption of memory if I get this correctly?